### ISSUE TITLE
Running Fred: Hanging on loading screen

### ISSUE THREAD

**Issue by Firlaev-Hans** [ruffle-rs/ruffle#11152]
### Describe the bug

[SWF](https://github.com/lightspark/lightspark/files/4082319/running_fred.zip)
Figured I'd give the recently added initial Stage3D support some workout with this Unity3D game, which at least has its main menu working in Lightspark.
In Ruffle, a window appears that renders the first frame of the loading screen, but it just freezes at that point. The following error is repeated in the terminal every ~2 seconds:
`ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x5557b35b5138 }) to handler FunctionObject(FunctionObject { ptr: 0x5557b02b7768 }) : TypeError: Error #1009: Cannot access a property or method of a null object reference. (accessing field: $Type)`

### Expected behavior

The game should load normally

### Affected platform

Desktop app

### Operating system

Fedora Linux 38 / KDE Plasma Wayland

**Comment by Lord-McSweeney** [comment]
This currently needs `flash.display3D.textures.Texture.uploadFromByteArray()`.

**Comment by Lord-McSweeney** [comment]
Thanks to #13180, this works now, but it takes _way_ too long to start up (more than a minute) and is very laggy at the title screen (about 0.5 fps for me). I pressed Space to get started, but I terminated the Ruffle process after waiting on the loading screen for 2-3 minutes.

**Comment by Firlaev-Hans** [comment]
> I pressed Space to get started, but I terminated the Ruffle process after waiting on the loading screen for 2-3 minutes.

I let it running for a bit, this was the result:
```
thread 'main' panicked at 'Failed to parse ATF texture: Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }', core/src/avm2/globals/flash/display3D/textures/texture.rs:81:14
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:67:14
   2: core::result::unwrap_failed
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/result.rs:1651:5
   3: ruffle_core::avm2::globals::flash::display3D::textures::texture::do_compressed_upload
   4: ruffle_core::avm2::globals::flash::display3D::textures::texture::upload_compressed_texture_from_byte_array
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
AVM2 stack trace: 
        at flash.display3D.textures::Texture/flash::display3D::textures::Texture::uploadCompressedTextureFromByteArray()
        at com.unity::UnityNative$/Ext_Stage3D_Upload_TextureAtf()
        at com.unity::UnityNative$/_ZN9GfxDevice15UploadTexture2DE9TextureID16TextureDimensionPhiiimibi16TextureUsageMode17TextureColorSpace()
        at com.unity::UnityNative$/_ZN9Texture2D13UploadTextureEb()
        at com.unity::UnityNative$/_ZN9Texture2D13AwakeFromLoadE17AwakeFromLoadMode()
        at com.unity::UnityNative$/_ZN18AwakeFromLoadQueue30PersistentManagerAwakeFromLoadE17AwakeFromLoadModePFvR6ObjectRK8TypeTreeE()
        at com.unity::UnityNative$/_ZN21PreloadLevelOperation7PerformEv()
        at com.unity::UnityNative$/_ZN14PreloadManager26UpdatePreloadingSingleStepEv()
        at com.unity::UnityNative$/_Z10PlayerLoopbb()
        at com.unity::UnityNative$/NativeExt_PlayerLoop()
        at com.unity::UnityContentInternal/playerLoop()
        at com.unity::UnityContent/onEnterFrame()
```

**Comment by Firlaev-Hans** [comment]
Now it manages to get in-game and sound works, it's just running _really_ slowly, with minute long loading times, multiple seconds per frame, and high CPU usage

**Comment by Aaron1011** [comment]
I'm closing this issue - the performance problems are a known issue and can be tracked in a separate ticket.

**Comment by Firlaev-Hans** [comment]
Note that, starting with 4bfd2fc978d5097517ffce53c42ee467c1339f8c, `--features jpegxr` must be specified, otherwise the game only shows a black screen.
